mcs/class/corlib/Test/System.IO/BinaryWriterTest.cs

```diff
@@ -419,6 +419,22 @@ public void WriteDecimal ()
 		Assert.AreEqual (0, bytes [45], "test#47");
 		Assert.AreEqual (0, bytes [46], "test#48");		
 	}
+
+	[Test]
+	public void WriteDecimalAndReadBack()
+	{
+		Decimal writeDec = -18295873486192640;
```

Comment:
```console
$ printf "0x%x\n" -18295873486192640
0xffbf000000000000
```
not sure what this value looks like in `Decimal` representation in memory, but I assume it will have a lot of null bytes. Any reason you chose this value?

Reply:
This is the same value used for BinaryReaderTest.ReadDecimal; if the byte order of the ints are reversed, it'll throw because the representation is then invalid.